src/extensions/default/bramble-move-file/MoveToDialog.js

```diff
@@ -114,9 +114,7 @@ define(function (require, exports, module) {
             MoveUtils.move(source, destination)
             .done(_finishMove.bind(null, source, destination))
             .fail(_failMove.bind(null, source, destination))
-            .always(function() {
-                dialog.close();
-            });
+            .then(dialog.close());
```

Comment:
@dangkhue27 what we do in `done` is use `.bind` to generate a new function. All of these,`done`, `then`, `fail`, etc. expect the parameter passed in to be a function (or a promise). When you call `dialog.close()`, it will use whatever is returned by calling that function and pass that as a parameter to `.then`. As you can see [here](https://github.com/mozilla/brackets/blob/master/src/widgets/Dialogs.js#L243-L247), the `close` function does not return anything, which evaluates to `undefined`. So really, what you're passing in is `.then(undefined)`.

To understand what's happening, try this experiment:
Remove the `always` code, change `.then(dialog.close())` to `dialog.close()` (without the `.then`). It should have the same effect.